.github/workflows/run_tests.yml

```diff
@@ -58,13 +58,15 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04] # cannot test on macOS as docker isn't supported on Mac
+        os: [ubuntu-22.04] # cannot test on macOS as docker isn't supported on Mac
         rai_v: [1.2.7] # versions of RedisAI
-        py_v: ['3.7.x', '3.8.x', '3.9.x', '3.10.x'] # versions of Python
-        compiler: [intel, 8, 9, 10, 11] # intel compiler, and versions of GNU compiler
+        py_v: ['3.8.x', '3.9.x', '3.10.x'] # versions of Python
+        compiler: [nvhpc-23-11, intel, gcc-11, gcc-12] # intel compiler, and versions of GNU compiler
+        link_type: [Static, Shared]
     env:
-      FC: gfortran-${{ matrix.compiler }}
-      GCC_V: ${{ matrix.compiler }} # used when the compiler is gcc/gfortran
+      COMPILER: ${{ matrix.compiler }} # used when the compiler is gcc/gfortran
```

Comment:
Are we going to need to differentiate like this for NVIDIA as well?

Reply:
The specification of these versions is now done within the setup section of each compiler chain. The only one that doesn't this to be done is the NVHPC compiler chain because it is installed as a single package (and not individually like Intel and GCC).